zio-http/src/main/scala/zhttp/service/ChannelFactory.scala

```diff
@@ -15,16 +15,17 @@ object ChannelFactory {
 
   object Live {
     def nio: UIO[jChannel.ChannelFactory[jChannel.Channel]] =
-      UIO(() => new jSocket.nio.NioSocketChannel())
+      UIO(() => new jSocket.nio.NioSocketChannel()).asInstanceOf[UIO[jChannel.ChannelFactory[jChannel.Channel]]]
 
     def epoll: UIO[jChannel.ChannelFactory[jChannel.Channel]] =
-      UIO(() => new jChannel.epoll.EpollSocketChannel())
+      UIO(() => new jChannel.epoll.EpollSocketChannel()).asInstanceOf[UIO[jChannel.ChannelFactory[jChannel.Channel]]]
 
     def kQueue: UIO[jChannel.ChannelFactory[jChannel.Channel]] =
-      UIO(() => new jChannel.kqueue.KQueueSocketChannel())
+      UIO(() => new jChannel.kqueue.KQueueSocketChannel()).asInstanceOf[UIO[jChannel.ChannelFactory[jChannel.Channel]]]
 
     def embedded: UIO[jChannel.ChannelFactory[jChannel.Channel]] =
       UIO(() => new jEmbedded.EmbeddedChannel(false, false))
+        .asInstanceOf[UIO[jChannel.ChannelFactory[jChannel.Channel]]]
```

Comment:
```suggestion
      UIO(new jEmbedded.EmbeddedChannel(false, false))
```

Reply:
I'm working on figuring out why the scala 3 compile is complaining without the `instanceOf` on all of these